monkey/infection_monkey/network/info.py

```diff
@@ -35,60 +38,72 @@ class NetworkAddress:
-
-        for line in f.readlines()[1:]:
-            iff, dst, gw, flags, x, x, x, msk, x, x, x = [var.encode() for var in line.split()]
-            flags = int(flags, 16)
-            if flags & RTF_UP == 0:
-                continue
-            if flags & RTF_REJECT:
-                continue
-            try:
-                ifreq = ioctl(s, SIOCGIFADDR, struct.pack("16s16x", iff))
-            except IOError:  # interface is present in routing tables but does not have any
-                # assigned IP
-                ifaddr = "0.0.0.0"
-            else:
-                addrfamily = struct.unpack("h", ifreq[16:18])[0]
-                if addrfamily == socket.AF_INET:
-                    ifaddr = socket.inet_ntoa(ifreq[20:24])
-                else:
-                    continue
-            routes.append(
-                (
-                    socket.htonl(int(dst, 16)) & 0xFFFFFFFF,
-                    socket.htonl(int(msk, 16)) & 0xFFFFFFFF,
-                    socket.inet_ntoa(struct.pack("I", int(gw, 16))),
-                    iff,
-                    ifaddr,
-                )
+    routes = []
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    ifreq = ioctl(s, SIOCGIFADDR, struct.pack("16s16x", LOOPBACK_NAME))
+    addrfamily = struct.unpack("h", ifreq[16:18])[0]
+    if addrfamily == socket.AF_INET:
+        ifreq2 = ioctl(s, SIOCGIFNETMASK, struct.pack("16s16x", LOOPBACK_NAME))
+        mask = socket.ntohl(struct.unpack("I", ifreq2[20:24])[0])
+        destination = socket.ntohl(struct.unpack("I", ifreq[20:24])[0]) & mask
+        ifaddress = socket.inet_ntoa(ifreq[20:24])
+        routes.append((destination, mask, "0.0.0.0", LOOPBACK_NAME, ifaddress))
+
+    for line in _read_route_file():
+        dst: bytes
+        msk: bytes
+        iff, dst, gw, flags, msk = _extract_network_info(line)
+        if flags & RTF_UP == 0:
+            continue
+        if flags & RTF_REJECT:
+            continue
+        ifaddr: Optional[str] = _get_interface_address(s, iff)
+        if ifaddr is None:
+            continue
+        routes.append(
+            (
+                socket.htonl(int(dst, 16)) & 0xFFFFFFFF,
+                socket.htonl(int(msk, 16)) & 0xFFFFFFFF,
+                socket.inet_ntoa(struct.pack("I", int(gw, 16))),
+                iff,
+                ifaddr,
             )
-
-        f.close()
-        return routes
+        )
+
+    return routes
+
+
+def _read_route_file() -> List[str]:
+    try:
+        with open("/proc/net/route", "r") as f:
+            return f.readlines()[1:]
+    except IOError:
+        return []
+
+
+def _extract_network_info(line: str) -> Tuple[bytes, bytes, bytes, int, bytes]:
+    values = [var.encode() for var in line.split()]
+    iff: bytes = values[0]
+    dst: bytes = values[1]
+    gw: bytes = values[2]
+    flags: int = int(values[3], 16)
+    msk: bytes = values[7]
+    return iff, dst, gw, flags, msk
```

Comment:
Another good candidate for a dataclass. But we don't have to go overboard with this refactoring. Our primary goal is cyclomatic complexity, which looks to be resolved.